storage/range.go

```diff
@@ -387,11 +387,16 @@ func (r *Range) redirectOnOrAcquireLeaderLease(timestamp proto.Timestamp) error
 	return nil
 }
 
-// verifyLeaderLease checks whether the requesting replica (by raft
-// node ID) holds the leader lease covering the specified timestamp.
-func (r *Range) verifyLeaderLease(originRaftNodeID proto.RaftNodeID, timestamp proto.Timestamp) bool {
-	l := r.getLease()
-	return originRaftNodeID == l.RaftNodeID && timestamp.Less(l.Expiration)
+// verifyLeaderLease checks whether the requesting replica (by raft node ID)
+// holds the leader lease covering the specified timestamp.
+func verifyLeaderLease(lease *proto.Lease, replicaRaftNodeID proto.RaftNodeID, timestamp proto.Timestamp) (bool, bool) {
```

Comment:
I'll give this (or something like this) a try, though `redirectOnOrAcquireLeaderLease` actually needs both the information about the lease holder and the expiry separately.